usr/lib/pkcs11/ep11_stdll/ep11_specific.c

```diff
@@ -243,6 +244,84 @@ static CK_RV check_cps_for_mechanism(cp_config_t * cp_config,
 static CK_RV get_control_points(STDLL_TokData_t * tokdata,
                                 unsigned char *cp, size_t * cp_len);
 
+/* Definitions for libica.
+ * Derived from ica_api.h so that there is no direct build dependency on libica
```

Comment:
why not add the libica dependency. i mean libica functionality is used , even per default.

Reply:
The code would build also with older header files (you dont need the new sha api prototypes). The new symbols could not be resolved but thats okay. Im in favor of adding an ep11 tok dependency to libica-devel package e.g. ica_api header.

Reply:
If I remember correctly, I tried to include ica_api.h first, but this lead to many compile errors, because ica_api.h caused many things to be redefined. E.g. ica_api.h includes <stdbool.h> which defines true/false. This conflicts with variables named true/false in OCK code. This of course could be changed in OCK code, but that would be a larger patch with lots of somehow unrelated changes.
Also unresolved symbols might be a problem, since OCK build/link fails when there are unresolved symbols.

Reply:
And finally, the ica_api.h header only defines the function prototypes, but not the typedefs for the function pointers. So those would have to be 'copied' anyway....

Reply:
Opened PR https://github.com/opencryptoki/opencryptoki/pull/137 to fix the 'true' issue.

The typedefs are needed anyway. Its only about the structure definitions (the problem would not exist if the struct defns were opaque).

How can the build fail with unresolved symbols when libica is dlopen'ed. I mean does it resolve symbols it does not use ?

Reply:
OK, I have rebased all patches on top of the last to PRs, and have changed the code to include ica_api.h to get rid of the definitions contained in ica_api.h.